src/lxc/conf.c

```diff
@@ -630,6 +631,30 @@ unsigned long add_required_remount_flags(const char *s, const char *d,
 #endif
 }
 
+static int add_shmount_to_list(struct lxc_conf *conf) {
+	char new_mount[MAXPATHLEN];
+	size_t len_mount;
+	/* Offset for the leading '/' since the path_cont
+	 * is absolute inside the container */
+	int ret = -1, offset = 1;
+
+	/* +1 for the separating whitespace */
+	len_mount = strlen(conf->lxc_shmount.path_host) + 1
+			+ strlen(conf->lxc_shmount.path_cont) - offset
+			+ sizeof(" none bind,create=dir 0 0") - 1;
+
+	ret = snprintf(new_mount, len_mount + 1, "%s %s none bind,create=dir 0 0",
```

Comment:
you never ensured len_mount < MAXPATHLEN

Reply:
According to `snprintf(3)`, it returns the number of bytes which _would_ have been printed if the buffer had been large enough, so maybe skip the length calculation, pass `sizeof(new_mount)` and verify the result is < that?